pkg/splunk/enterprise/configuration.go

```diff
@@ -708,6 +708,11 @@ func updateSplunkPodTemplateWithConfig(client splcommon.ControllerClient, podTem
 		})
 	}
 
+	// Add extraEnv from the CommonSplunkSpec config to the extraEnv variable list
+	for _, envVar := range spec.ExtraEnv {
+		extraEnv = append(extraEnv, envVar)
+	}
+
```

Comment:
Looking at the CRD changes seems like `[]corev1.EnvVar`, is supporting multiple ways of specifying the env variables to source from(configMap, secret object, etc..).  Wondering if we need to deal with them the same way, like reading those resource objects, and populating them as part of the extraEnv here?

Reply:
Since the extraEnv is the same `[]corev1.EnvVar` object, it will behave the same as all the other `EnvVar` objects.  Here we are just appending the `ExtraEnv` section to the env vars already created.  If `EnvVarSource` is used and the source doesn't exist, this should throw an error when creating the pod.
Applied CR:
```
apiVersion: enterprise.splunk.com/v1beta1
kind: Standalone
metadata:
  name: sa
  finalizers:
  - enterprise.splunk.com/delete-pvc
spec:
  replicas: 1
  extraEnv:
  - name: ANSIBLE_EXTRA_FLAGS
    value: "-vvv"
  - name: FROM_SOURCE
    valueFrom:
      secretKeyRef:
        name: doesntexistsecret
        key: awsSecretAccessKey
```
Error:
```
splunk-sa-standalone-0             0/1     CreateContainerConfigError   0          101s   10.244.0.22   kind-control-plane   <none>           <none>
...
Events:
  Type     Reason     Age               From               Message
  ----     ------     ----              ----               -------
  Normal   Scheduled  13s               default-scheduler  Successfully assigned default/splunk-sa-standalone-0 to kind-control-plane
  Normal   Pulled     1s (x3 over 13s)  kubelet            Container image "docker.io/splunk/splunk:8.1.3" already present on machine
  Warning  Failed     1s (x3 over 13s)  kubelet            Error: secret "doesntexistsecret" not found
```
I'm not sure we really need to this prior to apply since its already handled by kubectl.

Reply:
Thanks Jeff. 
Doing it prior may help in the case of incremental config change, where we destroy an existing Pod, before starting a pod with the missing resource, which may be a rare case :-)